inference-engine/src/vpu/graph_transformer/src/model/model.cpp

```diff
@@ -1952,8 +1952,6 @@ void ModelObj::removeStage(const Stage& stage) {
 void ModelObj::cleanUp() {
     for (const auto& data : datas()) {
         if (data->_usage == DataUsage::Input) {
-            VPU_THROW_UNLESS(!data->_consumerEdges.empty() || !data->childDataToShapeEdges().empty(),
-                    "Input data {} must either have at least one consumer (but got zero) or be a shape data.", data->name());
```

Comment:
Should we replace this assert with new one instead of removing it? I believe not all cases when data object has no consumers nor child data -> shape edges are considered possible

Reply:
I personally don't see any replacement for this assertion, because this assertion was intended to check, that there are no unused inputs, while now they are allowed and supported.

Reply:
They are allowed and supported only if they have corresponding outputs, right? Is it possible to have a check on that?

Reply:
In case if they have corresponding output, there will be at least one consumer, e.g. `Copy` from input to output. We also support the case where input doesn't have consumers at all, which, for example, may come from the case `Parameter->ShapeOf->Result` which is constant folded to `Parameter, Const->Result`.